providers/section.rb

```diff
@@ -28,7 +28,7 @@
 action :lock do
   @current_resource.exists = new_value?(doc.root, 'CONFIG/@overrideMode', 'Deny')
 
-  if !@current_resource.exists
+  if @current_resource.exists
```

Comment:
I think the confusion occurred because we're using @current_resource.exists (which is meant to signify if the resource exists) to actually mean "is the resource unlocked".  We should probably clean this up (at some nebulous time in the future).